pkg/pillar/devicenetwork/devicenetwork.go

```diff
@@ -118,28 +129,31 @@ func VerifyDeviceNetworkStatus(status types.DeviceNetworkStatus,
 		if err != nil {
 			errStr := "Onboarding certificate cannot be found"
 			log.Infof("VerifyDeviceNetworkStatus: %s\n", errStr)
-			return false, errors.New(errStr)
+			return false, errors.New(errStr), intfErrMap
 		}
 		clientCert := &onboardingCert
 		tlsConfig, err = zedcloud.GetTlsConfig(zedcloudCtx.DeviceNetworkStatus,
 			serverName, clientCert, &zedcloudCtx)
 		if err != nil {
-			errStr := fmt.Sprintf("TLS configuration for talking to Zedcloud cannot be found: %s", err)
-
-			log.Infof("VerifyDeviceNetworkStatus: %s\n", errStr)
-			return false, errors.New(errStr)
+			err = fmt.Errorf("TLS configuration for talking to Zedcloud cannot be found: %s", err)
+			log.Infof("VerifyDeviceNetworkStatus: %s\n", err)
+			return false, err, intfErrMap
 		}
 	}
 	zedcloudCtx.TlsConfig = tlsConfig
 	for ix := range status.Ports {
 		err = CheckAndGetNetworkProxy(&status, &status.Ports[ix])
 		if err != nil {
-			errStr := fmt.Sprintf("GetNetworkProxy failed %s", err)
-			log.Errorf("VerifyDeviceNetworkStatus: %s\n", errStr)
-			return false, errors.New(errStr)
+			ifName := status.Ports[ix].IfName
+			err = fmt.Errorf("GetNetworkProxy: ifName: %s, failed %s",
+				ifName, err)
+			log.Errorf("VerifyDeviceNetworkStatus: %s\n", err)
+			intfErrMap[ifName] = types.NewErrorInfo(err, "")
+			return false, err, intfErrMap
 		}
 	}
-	cloudReachable, rtf, err := zedcloud.VerifyAllIntf(&zedcloudCtx, testURL, retryCount, 1)
+	cloudReachable, rtf, err, intfErrMap := zedcloud.VerifyAllIntf(
```

Comment:
Since you are replacing intfErrMap here (as opposed to updating) that map is useless in this function since you only return the single error from CheckAndGetNetworkProxy

Reply:
If I make the above change, the question arises - should I collect both error? Sounds like I should.
Will merge both the errors from this function..

Reply:
Done..